content/blog/2019/corda-responder-validation/index.md

```diff
@@ -0,0 +1,185 @@
+  override fun checkTransaction(stx: SignedTransaction) {
+    // no validation
+  }
+})
+```
+
+That is a code snippet from one of my other blog posts. Yes... I have sinned, and I am also a hypocrite. I have failed so you can learn from my mistakes! 👩‍🏫
+
+Instead, it should look something like:
+
+```kotlin
+@InitiatedBy(SendMessageFlow::class)
+class SendMessageResponder(private val session: FlowSession) : FlowLogic<SignedTransaction>() {
+
+  @Suspendable
+  override fun call(): SignedTransaction {
+    val stx = subFlow(object : SignTransactionFlow(session) {
+      // [LedgerTransaction.verify] is called before [checkTransaction]
+      override fun checkTransaction(stx: SignedTransaction) {
+        val message = stx.coreTransaction.outputStates.single() as MessageState
+        check(message.recipient == ourIdentity) { "I think you got the wrong person" }
+        check(!message.contents.containsSwearWords()) { "Mind your language" }
+        check(!message.contents.containsMemes()) { "Only serious messages are accepted" }
+        check(message.sender.name.organisation != "Nigerian Prince") { "Spam message detected" }
+      }
+    })
+    return subFlow(ReceiveFinalityFlow(otherSideSession = session, expectedTxId = stx.id))
+  }
+}
+```
+
+Obviously, most of the things being checked here are pretty random, but it does show off what I have been talking about.
+
+- The `recipient` is checked to ensure that the `MessageState` is intended for the node it is being sent to.
+
+  ```kotlin
+  check(message.recipient == ourIdentity) { "I think you got the wrong person" }
+  ```
+
+This is an essential step to ensure that transactions you receive are intended for you and the role you play in the current flow, or in general, within the network. Not doing this, will allow your node to accept anything it is asked to sign and record. Even if it has absolutely nothing to do with you.
+
+- A series of other rules are also included. Each additional criteria is contractually valid but is not a desired property of the state in the context of this flow.
+
+  ```kotlin
+  check(!message.contents.containsSwearWords()) { "Mind your language" }
+  check(!message.contents.containsMemes()) { "Only serious messages are accepted" }
+  check(message.sender.name.organisation != "Nigerian Prince") { "Spam message detected" }
+  ```
+
+  I highly doubt that checking whether a message contains swear words would be contained inside of a contract. A message is still generally valid if it includes them. But, for the written flow, maybe it is a polite flow. Where any mention of a rude word must be beeped out (just like on pre-watershed TV). The same general gist applies to the other two rules.
+
+## How to validate
+
+I have just shown you what sort of validation to include. What I did not do is explain how the code shown before leads to a transaction being rejected. You might have sussed it out, but let me be thorough and go through it properly.
```

Comment:
Changing "did not "-> "didn't" makes the sentence flow a little better in my opinion. Doesn't make a huge difference though.